fuse/data/utils/samplers.py

```diff
@@ -137,8 +142,18 @@ def __init__(
             elif self._mode == "approx":
                 self._balanced_class_weights = [1 / self._num_balanced_classes] * self._num_balanced_classes
 
+        if self._sampler:
+            if self._verbose:
+                print(f"BatchSamplerDefault got a sampler of type: {type(self._sampler).__name__}")
+            # Get all samples that sampler posses.
+            # In use in DDP strategy: each process runs on a different GPU with a different instance of 'DistributedSampler'.
+            #   The DistributedSampler(s) make sure that each GPU posses a different subset of the samplers to avoid overlaps.
+            items = [i for i in self._sampler]  # get all samples that sampler posses
+        else:
+            items = None  # equivalent to all samples in dataset
+
         # get balanced classes per each sample
-        collected_data = dataset.get_multi(None, **self._dataset_get_multi_kwargs)
+        collected_data = dataset.get_multi(items=items, **self._dataset_get_multi_kwargs)
```

Comment:
We should be careful in that case.
If a sampler is given we will get subset. the order (index) won't be aligned with the original dataset.
Double-check me though.

Reply:
Not sure that I understood you.

We can restrict to use only `DistributedSampler` instances.